browser/resources/brave_new_tab_page_refresh/components/app.tsx

```diff
@@ -20,6 +20,16 @@ export function App() {
   const [settingsView, setSettingsView] =
     React.useState<SettingsView | null>(null)
 
+  React.useEffect(() => {
+    let params = new URLSearchParams(location.search)
+    let settingsArg = params.get('openSettings')
+    if (settingsArg === null) {
+      return
+    }
+    setSettingsView('background')
+    history.pushState(null, '', '/')
```

Comment:
this just resets the url, right, so when the page is refreshed the dialog closes?

Probably worth using the new `Navigation` API - its easier to use than `history.push/replaceState`
https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API

Reply:
Nice!

Reply:
It looks like the navigation API is perfect for doing any kind of routing, but requires some setup. I'm thinking we might want to set up (proper) routing scheme, and then we can remove this `openSettings` param once the old NTP goes away.